### ISSUE TITLE
x-axis baseline-shift different in html, svg, pdf

### ISSUE THREAD

**Issue by dereneaton** [sandialabs/toyplot#154]
The baseline-shift value does not seem to give consistent spacing between different outputs for the x-axis. Using Toyplot v.0.11.0 the following two figures saved as svg (top) and html (bottom) do not line up. The pdf output was also similar to svg. 

Here's my code:
```
def SNP_position_plot(distvar, distpis):
    ## make a canvas
    canvas = toyplot.Canvas(width=800, height=300)

    ## make axes
    axes = canvas.axes(xlabel="Position along RAD locus",
                       ylabel="N variables sites",
                       gutter=65)
    ## x-axis
    axes.x.ticks.show = True
    axes.x.label.style = {"baseline-shift":"-40px", "font-size":"16px"}
    axes.x.ticks.labels.style = {"baseline-shift":"-5px", "font-size":"12px"}
    axes.x.ticks.below = 5
    axes.x.ticks.above = 0
    axes.x.domain.max = maxend
    axes.x.ticks.locator = toyplot.locator.Explicit(
        range(0, maxend, 5), 
        map(str, range(0, maxend, 5)))
    
    ## y-axis
    axes.y.ticks.show=True
    axes.y.label.style = {"baseline-shift":"40px", "font-size":"16px"}
    axes.y.ticks.labels.style = {"baseline-shift":"5px", "font-size":"12px"}
    axes.y.ticks.below = 0
    axes.y.ticks.above = 5

    ## add fill plots
    x = np.arange(0, maxend+1)
    f1 = axes.fill(x, distvar, color=colormap[0], opacity=0.5)
    f2 = axes.fill(x, distpis, color=colormap[1], opacity=0.5)

    ## add a horizontal dashed line at the median Nsnps per site
    axes.hlines(np.median(distvar), opacity=0.9, style={"stroke-dasharray":"4, 4"})
    axes.hlines(np.median(distpis), opacity=0.9, style={"stroke-dasharray":"4, 4"})
    
    return canvas
```
![](https://dl.dropboxusercontent.com/u/2538935/toyplot_xaxis.png)

**Comment by tshead2** [comment]
OK, there are multiple issues here:

* We use the `dominant-baseline` attribute in the generated SVG to specify the text baseline.  This works fine in Safari, Firefox, and Chrome, but other apps (like Inkscape) ignore it completely, which can cause the output to look different depending on what you use to view it.  The Y axis doesn't appear to be affected because it uses an alphabetic baseline, which is the default.
* The PDF output can look different from the HTML output because PDF doesn't really have a concept of baselines, we have to adjust the text anchor explicitly instead, using whatever metrics we can get on the font.  There's no way to guarantee that this will produce the same result as `dominant-baseline` in a particular browser.

**Comment by tshead2** [comment]
It looks like the least-worst option is to stop relying on `dominant-baseline` and adjust the text anchor explicitly in the HTML backend.  That can guarantee that all outputs render the same, modulo subtle differences in font-rendering.  The downside is that the HTML does not have access to font metrics when the plot is generated, so the results / spacing / etc. are less likely to look good, and may require more manual tweaking of baselines by users.

**Comment by tshead2** [comment]
This is fixed in master, will be released in Toyplot 0.14.0.

Many thanks!
Tim
